massa-pos-worker/src/draw.rs

```diff
@@ -71,14 +72,13 @@ pub(crate) fn perform_draws(
             .map(|_index| addresses[dist.sample(&mut rng)])
             .collect();
 
+        let selection = Selection {
+            producer,
+            endorsements,
+        };
+        five_first_slots.push((cur_slot, selection.clone()));
```

Comment:
this will not only add the first 5 but all of them

Reply:
Fixed.